src/Magnum/DartIntegration/Object.h

```diff
@@ -40,72 +45,93 @@ namespace dart { namespace dynamics {
     class ShapeNode;
 }}
 
-namespace Magnum { namespace DartIntegration {
+namespace Magnum {
+
+class Mesh;
+class Buffer;
+template<UnsignedInt dimensions> class Texture;
+
+namespace DartIntegration {
+
+/**
+@brief Shape data
+
+@see @ref Object::_convertShapeNode()
+*/
+struct ShapeData {
+    /** @brief Mesh */
+    /* @todo: support multiple meshes */
+    Mesh* mesh;
+
+    /** @brief vertex Buffer */
+    Buffer* vertexBuffer;
+
+    /** @brief index Buffer */
+    Buffer* indexBuffer;
+
+    /** @brief Material data */
+    /* @todo: support multiple materials */
+    Trade::PhongMaterialData material;
+
+    /** @brief Textures */
+    Containers::Array<Texture<2>*> textures;
```

Comment:
Continuing with the forward declaration header from above, you could then simply use `Texture2D` here.